packages/paste-core/components/badge/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,597 @@
+    describe('Render', () => {
+      it('should render correctly when variant is "default"', () => {
+        const {getByTestId} = render(
+          <Badge data-testid="badge-1" variant="default">
+            test
+          </Badge>
+        );
+        const badgeElement = getByTestId('badge-1');
+
+        expect(badgeElement).toHaveStyleRule('color', 'colorTextWeak');
+        expect(badgeElement).toHaveStyleRule('background-color', 'colorBackground');
+      });
+
+      it('should render correctly when variant is "new"', () => {
+        const {getByTestId} = render(
+          <Badge data-testid="badge-2" variant="new">
+            test
+          </Badge>
+        );
+        const badgeElement = getByTestId('badge-2');
+
+        expect(badgeElement).toHaveStyleRule('color', 'colorTextNew');
+        expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundNew');
+      });
+
+      it('should render correctly when variant is "info"', () => {
+        const {getByTestId} = render(
+          <Badge data-testid="badge-3" variant="info">
+            test
+          </Badge>
+        );
+        const badgeElement = getByTestId('badge-3');
+
+        expect(badgeElement).toHaveStyleRule('color', 'colorTextNeutral');
+        expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundNeutralWeakest');
+      });
+
+      it('should render correctly when variant is "warning"', () => {
+        const {getByTestId} = render(
+          <Badge data-testid="badge-4" variant="warning">
+            test
+          </Badge>
+        );
+        const badgeElement = getByTestId('badge-4');
+
+        expect(badgeElement).toHaveStyleRule('color', 'colorTextWarningStrong');
+        expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundWarningWeakest');
+      });
+
+      it('should render correctly when variant is "error"', () => {
+        const {getByTestId} = render(
+          <Badge data-testid="badge-5" variant="error">
+            test
+          </Badge>
+        );
+        const badgeElement = getByTestId('badge-5');
+
+        expect(badgeElement).toHaveStyleRule('color', 'colorTextError');
+        expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundErrorWeakest');
+      });
+
+      it('should render correctly when variant is "success"', () => {
+        const {getByTestId} = render(
+          <Badge data-testid="badge-6" variant="success">
+            test
+          </Badge>
+        );
+        const badgeElement = getByTestId('badge-6');
+
+        expect(badgeElement).toHaveStyleRule('color', 'colorTextSuccess');
+        expect(badgeElement).toHaveStyleRule('background-color', 'colorBackgroundSuccessWeakest');
+      });
+    });
+  });
+
+  describe('Accessibility', () => {
+    it('Should have no accessibility violations', async () => {
+      const {container} = render(
+        <Badge data-testid="badge-1" variant="default">
```

Comment:
Ideally we'd have a as button and as anchor example in here too otherwise we're only testing the single variant for accessibility.

(I'm am aware that our coverage in other components is also like this and needs fixing)

Reply:
Yeah, I have some ideas about unit testing that might help us get better coverage without the heavy lift of many many unique `describe` and `it` blocks in test files (which can be laborious to create and maintain). I'll probably work up an example and share in the next show and tell.